docs/specs/component.md

```diff
@@ -205,53 +199,25 @@ sending it, like the `prometheus_exporter` sink, SHOULD NOT publish this metric.
-    defined properties as key-value pairs. It MUST NOT be rate limited.
+    defined properties as key-value pairs.
+  * MUST NOT be rate limited.
+
+#### ComponentError
 
-#### Error
+**Extends the [Error event].**
 
-*All components* MUST emit error events when an error occurs, and errors MUST be
-named with an `Error` suffix. For example, the `socket` source emits a
-`SocketReceiveError` representing any error that occurs while receiving data off
-of the socket.
+*All components* MUST emit error events in accordance with the [Error event]
+requirements.
 
 This specification does not list a standard set of errors that components must
 implement since errors are specific to the component.
 
-* Properties
-  * `message` - A human readable error message.
-  * `error_code` - An error code for the failure, if applicable.
-
-    `error_code` SHOULD only be specified if it adds additional information
-    beyond `error_type`.
-
-    The values for `error_code` for a given error event MUST be a bounded set
-    with relatively low cardinality because it will be used as a metric tag.
-    Examples would be syscall error code. Examples of values that should not be
-    used are raw error messages from `serde` as these are highly variable
-    depending on the input. Instead, these errors should be converted to an
-    error code like `invalid_json`.
-  * `error_type` - The type of error condition. This MUST be one of the types
-    listed in the `error_type` enum list in the cue docs.
-  * `stage` - The stage at which the error occurred. This MUST be one of
-    `receiving`, `processing`, or `sending`.
-  * If any of the above properties are implicit to the specific error
-    type, they MAY be omitted from being represented explicitly in the
-    event fields. However, they MUST still be included in the emitted
-    logs and metrics, as specified below, as if they were present.
-* Metrics
-  * MUST increment the `component_errors_total` counter by 1 with the defined
-    properties, except `message` as metric tags.
-  * MUST increment the `component_discarded_events_total` counter by the number
-    of Vector events discarded if the error resulted in discarding (dropping)
-    acknowledged events. For sources, only increment this metric if incoming
-    events were consumed (and acknowledged if applicable) and discarded. The
-    metric MUST not include events that will be re-ingested. For sinks, this
-    means only incrementing this metric if the error resulted in the sink
-    dropping the events, and thus acknowledging them. Retried events MUST not be
-    included in the metric.
-* Logs
-  * MUST log a message at the `error` level with the defined properties
-    as key-value pairs. It SHOULD be rate limited to 10 seconds.
+#### ComponentEventsDropped
+
+**Extends the [EventsDropped event].**
+
+*All components* that can drop events MUST emit a `ComponentEventsDropped`
+event in accordance with the [EventsDropped event] requirements.
```

Comment:
Shifted the `*EventsDropped` event requirements to the instrumentation spec since this is shared across Vector.